Fix for Django 18 and typo correction

rcooke opened the pull request:
I'm creating this pull request just to bring two things to you attention:
- There is a typo here:  msg = _(u"Locked by %s for %s longer") % (until, locked_by_name)
- It should be:  msg = _(u"Locked by %s for %s longer") % (locked_by_name, until)
- I had to change some undocumented object names to get this to run under Django 18, as per:
  - https://code.djangoproject.com/ticket/20853

rcooke committed:
Update PIP install url

jaylett commented:
(a) you should look at the [`pando-fork` branch](https://github.com/jaylett/django-locking/tree/pando-fork), which certainly works with Django 1.7 migrations, and may be ready for 1.8 (I haven't tried it, but I've fixed the warnings that 1.7 gives about future deprecation; looking at your changes, the `module_name` -> `model_name` I seem to have missed so far, probably because the don't appear on `manage.py check` / startup)
(b) if you don't like the other stuff I've done, you probably want to PR against theatlantic's upstream repo
(c) I like the verbose logging; maybe make the logger configurable and default to `django.locking` or something?
(d) more, smaller, commits would enable me to merge parts of this more readily, if you do want this merged into my fork :-)

I'll leave this PR open in case you want to change things and force push to your branch. Thanks!

rcooke committed:
Fix order of locker and time remaining

The hover-over message for the lock icon had the position of
time remaining in front of the locking user.

rcooke commented:
Ok.  I'm getting my act together!  (finally).  I want to upgrade to Django 1.10, install Grapelli, and investigate issues with locking not working on Android devices.  To start with, I'm breaking my previous multi-commit down into singles like I should have to begin with - the problem with migrating from Subversion....

Here is just the bug fix in your code.  You will want this one.

Yes, I will be cherry picking from some of the other branches as well.

The pando-fork appears to be a clone of your v2 head?  I have not looked at your other branches, anything of particular interest in there?

Hm.  Github is still tacking all my commits into this pull request.  I was hoping to keep my internal stuff seperate so it would not get in your way.  Is there a manual to explain how to do that?  Or now that I am breaking them up into single commits, does that make it easy for you to cherry pick them?

rcooke committed:
Internal variable name change in Django 1.8

    As noted here:
    https://code.djangoproject.com/ticket/20853

    The name of an internal variable has changed in recent Django versions.
    This change was needed for Django 18.

rcooke committed:
Django 1.8 migration to add locking table

This migration creates the locking table needed by django-locking.

rcooke committed:
Make locking table visible for diagnostics

Add the table to the Admin interface, and create a
handy summary line for each lock record.  Very handy when
trying to figure out issues with locking.

rcooke committed:
Python anywhere logging setup

If you work with PythonAnywhere.com these logger names
are part of how one gets diagnostic messages into their log.

rcooke committed:
Remove trailing spaces

rcooke commented:
Aha!  I created a Django110 branch for my ugly logging commits and whatever else I get into.

I think you should pull all these changes, then comment out the lines added for making locking table visible, that way the line is there when needed, but not active by default.  I dunno what you want to do with migrations.  I love South, but since I created this project wihtout it, I'm sticking to Django's built in version.

jaylett commented:
I'm not working on the project that used this any more, which makes it difficult for me to unstash the information I need to be able to respond to this. Some immediate thoughts:
1. Grappelli doesn't support 1.10 yet
2. I am unlikely ever to use Grappelli again, so I'd be more likely to go back to the upstream (Atlantic) version in future and then forward port the fixes that are needed. I'm uncomfortable about the idea of maintaining (even very passively) yet another fork, which feels bad to me. (The Atlantic folks are friendly and helpful.)
3. I don't remember what the other branches are, so you can probably ignore them :-)
4. I initially tried to construct a PR for upstream, but it was a mess and I don't expect it to get merged without further work. I _believe_ that the South support was for their benefit, but I now can't remember.
5. Commits shouldn't contain commented-out code.
6. If pythonanywhere can't cope with configuring arbitrary loggers, then something is very wrong :-(

rcooke commented:
Thanks for taking the time to repsond, I really appreciate your input.

PythonAnywhere has a syslog service, if you configure a root logger to use it, then all your log messages go into your user account server log - which is very handy.  Both as a thread-safe thing, and as a one-place-to-see-everything-that-failed by remote thing.  It took me a while to figure out it works best if the root is called "project".  Then you instantiate loggers wherever you like, but as a child of "project".  That gets your stuff sysloged, and, structures the message so you can tell where it came from.  I have a few Django projects with 20 installed apps.  Having them all log as "admin" was not very helpful!

I'm on the Grapelli newslist, I thought they were talking about a 110 release.  I'll dig into that shortly. It won't kill me to stay with Django 1.9, I don't even know if the new featuers in 1.10 are of interest to this project.  I just like to stay current whenever I can - and its easy.

I'm more interested in looking into the reports that the Android Galaxy tablet the client uses is not respecting the locks very well.  Even better if this has already been addressed by The Atlantic.

I just pushed a merger of Atlantics head V2 into my Djano110 branch.  I did not see any conflicts with your "Grapelli Support" - although I'm not sure what that would be, guess I'll find out!

I'm scouting the other forks looking for gems to cherry pick...

jaylett commented:
My grapelli support, IIRC, was just making it use a more permissive check to find containers, so I'm not surprised that doesn't cause conflicts. Good news!

From my pov I already cherry-picked the other forks for the best bits (there are various commits in my tree which are authored by someone else then committed by me). However you may find other useful bits either that I missed or since then.

Maybe the logging target name should be configurable? I can see the value for you under PythonAnywhere, but I'd just drop stuff into Django's `LOGGING` configuration to do the right thing, and having a semantic name makes that much clearer.

My approach these days to staying current with Django is to use the LTS where I can (currently 1.8). If you're warnings-clean on that, the upgrade to the next LTS shouldn't be too painful (this is only really true as of 1.8 onwards). You basically get two years between upgrades that way unless you need newer features.

rcooke commented:
Yeah, yeah, I added your idea of making the logging name more configurable the first time you brought it up (when I first made this PR).  Its low on my todo list though.

I merged in changes from MotherNatureNetwork/v2-rebased.  Both it and TheAtlantic had changes after your fork and last commit.  Some minor fixes, some improvements, all good.

All the other forks based on the same fork you started with dont seem to have anything new - lots of duplication.  A few people may have made the same mistake I first did (a la SVN) to commit too many things at once.  Makes it too hard to dig out the gems.  I skipped them.  I have not really looked at any forks that have older ancestry, I figure they are likely not going to be easily merged.

User Digicase has a more comprehensive Grappelli support solution.   I also posted a question about django-locking support on Grappelli's project maintainer's repo.  I'm going to test what I have, and likely pull in Digicase's JS changes for Grappelli - after I verify they are needed.  His changes conflict with yours, I don't want to have to sort it out unless I have too.

When I was young and foolish, I upgraded all the time!  heheh  Now I don't budge until something forces me to.  I had a machine running Windows 3.11 for Workgroups until it was replaced in 2010....

By the way, The Atlantic might have a superior DJ18 update than the one I proposed in this PR.  I make this note here so if in the future you need this code this will remind you to look at the Django110 branch or at v2 after I merge back to it.
